src/components/stockVisualisationComponents/ChartTypes/PriceChart/PriceChart.js

```diff
@@ -10,45 +10,85 @@ import {
-import { useState } from "react";
+import { useState, useEffect, PureComponent } from "react";
 
-function StockPriceChart() {
+/// API ///
+import { APIName } from "../../../../constants/APIConstants";
+import { API } from "aws-amplify";
 
-    const month = [
-        { name: 'month', price: 100 },
-        { name: 'month', price: 500 },
-        { name: 'month', price: 60 },
-        { name: 'month', price: 700 },
-        { name: 'month', price: 800 }]
+
+class CustomizedAxisTick extends PureComponent {
+    render() {
+        const { x, y, stroke, payload } = this.props;
+
+        return (
+            <g transform={`translate(${x},${y})`}>
+                <text x={0} y={0} dy={16} textAnchor="end" fontSize={1} fill="#ffffff" transform="rotate(-35)">
+                    {payload.value}
+                </text>
+            </g>
+        );
+    }
+}
+
+function StockPriceChart({ symbol }) {
+
+    const [loading, setLoading] = useState(true);
+    const [oneMonthPrices, setOneMonthPrices] = useState();
+    const [oneYearPrices, setOneYearPrices] = useState();
+    const [error, setError] = useState("");
+
+
+    useEffect(() => {
+        /// getStockInfo ///
+        // Description:
+        //  Makes a GET request to the backend route /stock/OneYearStockData/:symbol
+        const getOneMonthPrices = async () => {
+            try {
+                // Request is being sent set loading true 
+                setLoading(true);
+                // Set the path and use symbol to get single stock
+                let path = `/api/stock/OneYearStockData/${symbol}`
+                // Send the request with API package
+                const res = await API.get(APIName, path);
+                let path2 = `/api/stock/OneMonthStockData/${symbol}`
+                const res2 = await API.get(APIName, path2);
+                setOneYearPrices(res);
+                setOneMonthPrices(res2);
+                setLoading(false);
+            } catch (error) {
+                // Log the error 
+                console.log(error);
+                // Set the error message to be displayed on the page 
+                setError(error.response.data.errormessage);
+                setLoading(false);
+            }
+        }
+        getOneMonthPrices();
+    }, [])
 
     const day = [
-        { name: 'day', price: 400 },
-        { name: 'day', price: 700 },
-        { name: 'day', price: 60 },
-        { name: 'day', price: 700 },
-        { name: 'day', price: 500 }]
-
-    const year = [
-        { name: '2017', price: 400 },
-        { name: '2018', price: 10 },
-        { name: '2019', price: 60 },
-        { name: '2020', price: 700 },
-        { name: '2021', price: 100 }]
+        { date: '2022-01-10', price: 400 },
+        { date: '2022-01-11', price: 700 },
+        { date: '2022-01-12', price: 60 },
+        { date: '2022-01-13', price: 700 },
+        { date: '2022-01-1', price: 500 }]
```

Comment:
Should these be hardcoded?

Reply:
They should be, we don't have daily data yet. My weekend job.

Reply:
Grand so! Wasn't positive